docs/proposals/fluentd-limit-requests-to-apiserver.md

```diff
@@ -0,0 +1,107 @@
+
+1. Moving the normalization off the compute nodes
+2. Reducing the number of open connections by utilizing a smaller number of normalizers to enrich all the logs in lieu of each Fluentd instance enriching its own logs.
+
+This will require:
+* Deployment changes (e.g. deployer or ansible)
+* ~~Packing and containerization of rsyslog~~
+  * Packaging and containerization tasks completed by CDA team
+
+Questions:
+1. Does this require any Fluentd plugin changes?
+  * A: syslog input filter exists but is not encrypted[3]
+2. Does this require modifications to Fluentd? (e.g. how it receives logs, ships logs)
+  * A: Only the input to Fluentd changes.  Everything downstream remains the same
+3. How many Fluentd instance are needed?
+  * A: TBD
+1. Where do Fluentd instance get deployed? Infra nodes only?
+2. Are there 'collection' plugins available to Fluentd where there are not equivalents to rsyslog thus limiting easy customization?
+
+### Option 3 - Fluentd (Collector) to Fluentd (Normalizer)
+This is similiar to **Option 2** but it utilizes Fluentd as the collector instead of rsyslog.  
+#### Limitations
+This scenerio becomes more complicated to scale and deploy because:
+* Multiple configmaps to maintain (e.g. gather fluentd, enrichment fluentd)
+* Potentially two daemonsets
+* Certain nodes potentially more resource intensive (e.g. cpu, network, memory)
+
+This will require:
+* Deployment Changes (e.g. deployer or ansible)
+
+Questions:
+See questions from **Option 2**
+
+### Option 4 - Metadata service
+Introduce a service to the deployment of EFK that is solely responsible for watching Pods and gathering metadata that is provided by the Kubernetes Metadata filter.  The service is queried by Fluentd to retrieve the required information.  The expectation is instead of having N Fluentd pods contacting the API server that there are some significantly less number of 'Metadata services' that provide the same information with reduced requests.
+
+This will require:
+* New Fluend plugin to interact with the new service
+* Creation and maintenance of the service code
+  * e.g. technology choice, github repo, deployment changes (ansible), enterprise repo, ci, packaging
+
+Questions:
+1. What scale is required? X instances per N cluster nodes
+1. Impact on cluster infrastructure
+1. Do instances need to be constrained to infra nodes?
+2. Is this option really add value over the collector to normalizer options?
+
+### Option 5 - Kubelet Handler for Pod metadata
+Add a handler to the Kubelet to allow pod information to be queried by name or to instantiate a watch.  Ideally, the endpoint would feature the same characteristics of the API master pod endpoint to allow reuse of the existing metadata filter [2].  This would completely eliminate connections to the master.
```

Comment:
@derekwaynecarr Are we likely receive pushback upstream to get this request because of that?